### PR TITLE
Replace all commands in NEU with a brigadier implementation

### PR DESCRIPTION
**Pull request by romangraef**
This adds more/better autocompletion, a nicer DSL for defining complex commands and help descriptions for every command, available via `/neuhelp command`

### CODE REVIEW

File: src/main/kotlin/io/github/moulberry/notenoughupdates/commands/misc/ProfileViewerCommands.kt [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+import net.minecraft.util.EnumChatFormatting.RED
+import net.minecraftforge.fml.common.Loader
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+@NEUAutoSubscribe
+class ProfileViewerCommands {
+    companion object {
+        fun CommandContext<ICommandSender>.openPv(name: String) {
+            if (!OpenGlHelper.isFramebufferEnabled()) {
+                reply("${RED}Some parts of the profile viewer do not work with OptiFine Fast Render. Go to ESC > Options > Video Settings > Performance > Fast Render to disable it.")
```
Comment (NopoTheGamer) [inline src/main/kotlin/io/github/moulberry/notenoughupdates/commands/misc/ProfileViewerCommands.kt:41]: Is this even true anymore?

Comment (romangraef) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/commands/misc/ProfileViewerCommands.kt:41]: I don't know.

Comment (romangraef) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/commands/misc/ProfileViewerCommands.kt:41]: Also out of scope of this PR

**Comment by hannibal002** [comment]
```
 java.lang.RuntimeException: Warning: Setting help on node that cannot be executed. Will be ignored
 	at io.github.moulberry.notenoughupdates.util.brigadier.BrigadierRoot.setHelpForNode(BrigadierRoot.kt:50)
 	at io.github.moulberry.notenoughupdates.util.brigadier.DslKt.withHelp(dsl.kt:162)
 	at io.github.moulberry.notenoughupdates.commands.dev.DiagCommand$onCommands$1$1.invoke(DiagCommand.kt:45)
 	at io.github.moulberry.notenoughupdates.commands.dev.DiagCommand$onCommands$1$1.invoke(DiagCommand.kt:39)
 	at io.github.moulberry.notenoughupdates.util.brigadier.DslKt.then(dsl.kt:143)
 	at io.github.moulberry.notenoughupdates.util.brigadier.DslKt.thenLiteral(dsl.kt:128)
 	at io.github.moulberry.notenoughupdates.commands.dev.DiagCommand$onCommands$1.invoke(DiagCommand.kt:39)
 	at io.github.moulberry.notenoughupdates.commands.dev.DiagCommand$onCommands$1.invoke(DiagCommand.kt:38)
 	at io.github.moulberry.notenoughupdates.util.brigadier.DslKt.literal(dsl.kt:122)
 	at io.github.moulberry.notenoughupdates.events.RegisterBrigadierCommandEvent.command(RegisterBrigadierCommandEvent.kt:43)
 	at io.github.moulberry.notenoughupdates.commands.dev.DiagCommand.onCommands(DiagCommand.kt:38)
 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_111_DiagCommand_onCommands_RegisterBrigadierCommandEvent.invoke(.dynamic)
 	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:49)
 	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:140)
 	at io.github.moulberry.notenoughupdates.events.NEUEvent.post(NEUEvent.java:27)
 	at io.github.moulberry.notenoughupdates.util.brigadier.BrigadierRoot.registerHooks(BrigadierRoot.kt:94)
 	at io.github.moulberry.notenoughupdates.util.brigadier.BrigadierRoot.updateHooks(BrigadierRoot.kt:82)
 	at io.github.moulberry.notenoughupdates.NotEnoughUpdates.preinit(NotEnoughUpdates.java:283)

```

That indended?

**Comment by romangraef** [comment]
Nope, that warning shouldnt happen

**Comment by NopoTheGamer** [comment]
/neusouls on/off/markfound/marknotfound doesnt work
Should all be prefixed - not sure if this happened before - https://cdn.discordapp.com/attachments/702456294874808330/1073567169016778842/image.png
lol - https://cdn.discordapp.com/attachments/702456294874808330/1073567468540403772/image.png
cant do /neupronouns <mcname> - https://cdn.discordapp.com/attachments/702456294874808330/1073568481427079179/image.png
/neulinks probably shouldnt show the prefix on the empty line - https://cdn.discordapp.com/attachments/702456294874808330/1073569488726929450/image.png

**Comment by romangraef** [comment]
neusouls is not covered by me. ill fix reload repository ig. oh. does /neupronouns minecraft throwpo still work? i can fix that if yes. neulinks i dont even know whats going on, im just gonna make neulinks completely ignore the prefix

**Review by hannibal002 — APPROVED** [review]

**Review by Lulonaut — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Pull request merged**